### PR TITLE
[workloads] Add new workloads for net8.0

### PR DESCRIPTION
**Pull request by lewing**
Backport the workload changes from net7 but and a versionless manifest that represents the netappcurrent so that version bumps will require fewer immediate changes in net9.0

- `Wasm.Build.Tests` installs `wasm-tools`, and `wasm-tools-net7`
- and uses a sdk with both `wasm-tools`, and `wasm-tools-net7` for running, with projects built for `net7.0`
- `net6.0` workload install is attempted, but fails because the `6.0.10` package isn't on public feeds yet

### CODE REVIEW

**Comment by lewing** [comment]
Net8 workloads still won't really work in wbt until there is a proper net8 sdk without another hack

File: src/mono/nuget/Microsoft.NET.Runtime.WebAssembly.Sdk/Sdk/AutoImport.props [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <Project>
   <PropertyGroup>
-    <WasmNativeWorkload>true</WasmNativeWorkload>
+    <WasmNativeWorkloadAvailableList>+net6.0+net7.0+net8.0+</WasmNativeWorkloadAvailableList>
```
Comment (lewing) [inline src/mono/nuget/Microsoft.NET.Runtime.WebAssembly.Sdk/Sdk/AutoImport.props:3]:
```suggestion
    <WasmNativeWorkloadAvailableList>+net6.0+net7.0+net8.0+</WasmNativeWorkloadAvailableList>
```
copy what release/7.0 does here not this

File: src/mono/nuget/Microsoft.NET.Workload.Mono.Toolchain.Manifest/WorkloadManifest.targets.in [hunk 2]
```diff
@@ -1,93 +1,113 @@
...
-        <UsingBrowserRuntimeWorkload Condition="'$(UsingBrowserRuntimeWorkload)' == ''" >$(WasmNativeWorkload)</UsingBrowserRuntimeWorkload>
+    <PropertyGroup Condition="'$(RuntimeIdentifier)' == 'browser-wasm' and '$(TargetFrameworkIdentifier)' == '.NETCoreApp'">
+      <_BrowserWorkloadNotSupportedForTFM Condition="$([MSBuild]::VersionLessThan($(TargetFrameworkVersion), '6.0'))">true</_BrowserWorkloadNotSupportedForTFM>
+      <_BrowserWorkloadDisabled>$(_BrowserWorkloadNotSupportedForTFM)</_BrowserWorkloadDisabled>
+
+      <!-- Is the workload for the target framework available -->
+      <WasmNativeWorkloadAvailable />
+      <WasmNativeWorkloadAvailable Condition="'$(TargetsNet8)' == 'true' and $(WasmNativeWorkloadAvailableList.Contains('+net8.0+'))">true</WasmNativeWorkloadAvailable>
+      <WasmNativeWorkloadAvailable Condition="'$(TargetsNet7)' == 'true' and $(WasmNativeWorkloadAvailableList.Contains('+net7.0+'))">true</WasmNativeWorkloadAvailable>
+      <WasmNativeWorkloadAvailable Condition="'$(TargetsNet6)' == 'true' and $(WasmNativeWorkloadAvailableList.Contains('+net6.0+'))">true</WasmNativeWorkloadAvailable>
```
Comment (lewing) [inline src/mono/nuget/Microsoft.NET.Workload.Mono.Toolchain.Manifest/WorkloadManifest.targets.in:22]: until we can update the AutoImports in the other branches this needs to check for versioned props

**Review by lewing — COMMENTED** [review]
the autoimport stuff can't work this way

**Review by radical — APPROVED** [review]

**Comment by radical** [comment]
`Installer Build and Test coreclr OSX_arm64 Release` - failed due to network issues.
`libraries Test Run release coreclr Linux x64 Debug` is https://github.com/dotnet/runtime/issues/75378 .
`Libraries Test Run release mono Linux arm64 Debug` - timed out, and it unrelated to this PR.
`mono llvmaot Pri0 Runtime Tests Run Linux x64 release` - https://github.com/dotnet/runtime/issues/75391

**Comment by radical** [comment]
All the failures are unrelated. Merging this, which will also unblock `main`.

**Pull request merged**